File: FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/request/node/BaseNodeRequest.java
package com.flowcrypt.email.api.retrofit.request.node;

import android.content.Context;
import android.net.Uri;

import com.flowcrypt.email.api.retrofit.node.NodeService;

import java.io.IOException;

import retrofit2.Response;

/**
 * It's a base request for the Node backend.
 *
 * @author Denis Bondarenko
 * Date: 1/10/19
 * Time: 5:43 PM
 * E-mail: DenBond7@gmail.com
 */
public class BaseNodeRequest implements NodeRequest {

  protected Context context;
  protected Uri uri;

  public BaseNodeRequest(Context context, Uri uri) {
    this.context = context != null ? context.getApplicationContext() : context;
    this.uri = uri;
  }

  public BaseNodeRequest() {
  }

  @Override
  public String getEndpoint() {
    throw new UnsupportedOperationException("not defined");
  }

  @Override
  public byte[] getData() {
    return new byte[0];
  }

  @Override
  public Uri getUri() {
    return uri;
  }

  @Override
  public Context getContext() {
    return context;
  }

  @Override
  public Response getResponse(NodeService nodeService) throws IOException {
    throw new UnsupportedOperationException("not defined");
  }
}

File: FlowCrypt/src/main/java/com/flowcrypt/email/api/retrofit/request/node/BaseNodeRequest.kt
/*
 * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
 * Contributors: DenBond7
 */

package com.flowcrypt.email.api.retrofit.request.node

import android.content.Context
import android.net.Uri
import com.flowcrypt.email.api.retrofit.node.NodeService
import retrofit2.Response

/**
 * It's a base request for the Node backend.
 *
 * @author Denis Bondarenko
 * Date: 1/10/19
 * Time: 5:43 PM
 * E-mail: DenBond7@gmail.com
 */
open class BaseNodeRequest constructor(
  final override var context: Context?,
  override val uri: Uri?,
  override val hasEncryptedDataInUri: Boolean = false
) : NodeRequest {

  init {
    this.context = context?.applicationContext
  }

  override val endpoint: String
    get() = throw UnsupportedOperationException("not defined")

  override val data: ByteArray
    get() = ByteArray(0)

  constructor() : this(null, null)

  override fun getResponse(nodeService: NodeService): Response<*> {
    throw UnsupportedOperationException("not defined")
  }
}
